### ISSUE TITLE
[vSphere] Waiting for SSH...

### ISSUE THREAD

**Issue by rmammadli** [rancher/rancher#35935]
**Rancher Server Setup**
- Rancher version: v2.6.2
- Installation option: Helm Chart
   - Kubernetes Cluster and version: RKE1
- Proxy/Cert Details: v1.6.1

**Information about the Cluster**
- Kubernetes version: 1.21.7
- Cluster Type: Downstream
   - Infrastructure Provider: vSphere

**Describe the bug**

Hi,
I am facing the problem with deploying workload cluster from rancher v2.6.2 using vm template.
Template is based on ubuntu 20.04 and has all requirements fulfilled, like cleaning cloud-init and installing open-vm-tools.
There is a need to configure static ip addresses for nodes. 
Unfortunately i am constantly getting the message 'waiting for ssh...' when i am using cloud_config in node template for node's network configuration.
Log entries in newly created node of downstream cluster show that docker user has been installed / ingested by rancher server on node(s) successfully.

**To Reproduce**
 
Use node template with cloud config to assign at least static ip address for provisioning user / workload cluster for infrastructure provider vSphere.
Note: without cloud config used and with static ip address preconfigured inside vm template, docker user from rancher server connects to node created successfully.

**Result**
2021/12/21 07:37:32 [INFO] [node-controller-rancher-machine] Waiting for SSH to be available...

**Expected Result**

Workload / downstream cluster would be deployed with the success.

**Screenshots**

![rancher](https://user-images.githubusercontent.com/24699861/146890980-605bbf0d-9c3f-4a9f-8166-ff1466b5b8ee.PNG)

**Additional context**

**Comment by caatclaudia** [comment]
Did you resolve this issue? I am the same problem but with OpenStack Integration.
I saw [information](https://rancher.com/docs/rke/latest/en/example-yamls/) that you can add this at your YAML Cluster:
```
bastion_host:
    address: x.x.x.x
    user: ubuntu
    port: 22
    ssh_key_path: /home/user/.ssh/bastion_rsa
```
But this doesn't solve my problem.

**Comment by rmammadli** [comment]
Hi @caatclaudia,
unfortunatelly i could not solve the issue by using cloud_config.

**Comment by caatclaudia** [comment]
When I had this error I was using a Cirros Image, but then I tried with Debian 10 and it worked. Maybe you could try with another Operating System.

**Comment by rmammadli** [comment]
We just try to use os version, which is already validated and certified:
https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-6-3/

**Comment by rmammadli** [comment]
Issue has been solved after upgrading vSphere to v7.0.x
